src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
Thx @pspill! So the current ordering is OK (settings > plugins > proxy), except for the fact that the imported providers need to be sorted in a deterministic way. 

In other words: the imported providers should be enumerated in the same order every single time to ensure a consistent experience. 
Perhaps it could be as simple as ordering them by type name ascending? Or do you need something more sophisticated using MEF Metadata? (any requirements that would drive this ordering?)